### ISSUE TITLE
Selected data source not updating the panel

### ISSUE THREAD

**Issue by sbhushankr** [grafana/grafana#50559]
<!--
I am facing the same issue. Like we have created the dashboard with Prometheus data source and created the variable as Data source and when we selecting the data source in variable but dashboard is updating and even panel query showing old selected data source there.

<img width="534" alt="Capture2" src="https://user-images.githubusercontent.com/44037104/172986163-625a5699-2349-4326-9571-5055d0af5e96.PNG">

**Environment**:  Dev  ,  Very soon will move on PROD
- Grafana version: 8
- Data source type & version:  Prometheus
- OS Grafana is installed on:  Linux
- User OS & Browser:  Chrome
- Grafana plugins:  
- Others:

Regards, 
Shashi

**Comment by sbhushankr** [comment]
Hi,
Can anyone faced this issue and have any solution then please share it here. I am using common grafana for many datasource and because of this issue we need to create the separate dashboard for each datasource. This become a big concern to manage so many dashbaord for same type of requirement.

Regards,
Shashi

**Comment by ashharrison90** [comment]
hi @sbhushankr 👋 

can you please fill out as much of the issue template as possible so we can help investigate your issue?

```
---
name: Bug report
about: Report a bug you found when using Grafana
labels: 'type: bug'
---

<!--
Please use this template to create your bug report. By providing as much info as possible you help us understand the issue, reproduce it and resolve it for you quicker. Therefore take a couple of extra minutes to make sure you have provided all info needed.

PROTIP: record your screen and attach it as a gif to showcase the issue.

- Questions should be posted to: https://community.grafana.com
- Use query inspector to troubleshoot issues: https://bit.ly/2XNF6YS
- How to record and attach gif: https://bit.ly/2Mi8T6K
-->

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:
```

thanks! 😄

**Comment by torkelo** [comment]
@sbhushankr what version of Grafana are you on? 

We have a fix in 8.5.4 https://github.com/grafana/grafana/pull/49478   that I think fixes your issue. But you might have to go into panel edit and select the data source again for the fix to take effect

**Comment by sbhushankr** [comment]
Hi, 

Following is my current version

- Grafana version:  V8.5.3
- Data source type & version:   Prometheus  (AWS Service) 2.34.0
- OS Grafana is installed on: RHEL 8.6
- User OS & Browser:  Windows 10, Chrome
- Grafana plugins: NA
- Others:  NA

Regards,
Shashi
